### PR TITLE
Minor changes to Amapa parser

### PR DESCRIPTION
**Pull request by AnyaLindstromBattle**
Changes to date function to deal with empty fields and minor changes for readability and/or consistency with other parsers

### CODE REVIEW

**Comment by iamleeg** [comment]
Looks like the test failure is due to the preexisting conditions list.

**Comment by AnyaLindstromBattle** [comment]
> Looks like the test failure is due to the preexisting conditions list.

yeah I always get encoding issues when there are accents involved - think it's due to Excel weirdness

File: ingestion/functions/parsing/brazil_amapa/amapa.py [hunk 1]
```diff
@@ -15,19 +15,19 @@
...
+_METHOD_CONFIRMATION = "tipoTeste"
+_CONFIRMATION = "resultadoTeste"
+_GENDER = "sexo"
+_NOTES_BAIRRO = "bairro"
+_MUNICIPALITY = "municipio"
+_AGE = "idade"
+_ETHNICITY = "racaCor"
+_NOTES_INDIGENOUS_GROUP = "etniaIndigena"
+
+_COMORBIDITIES_MAP = {
```
Comment (iamleeg) [inline ingestion/functions/parsing/brazil_amapa/amapa.py:30]: This is a great change, it will be more reliable if the columns get reordered in the data.

**Review by iamleeg — APPROVED** [review]
Looks good!

File: ingestion/functions/parsing/brazil_amapa/amapa.py (hunk 1 above)
Comment (AnyaLindstromBattle) [reply ingestion/functions/parsing/brazil_amapa/amapa.py:30]: yep, exactly

**Review by calremmel — APPROVED** [review]
Great changes - this should resolve the error I'm currently seeing in the log. Thanks, Anya!

**Comment by z023** [comment]
@AnyaLindstromBattle Amapa is still showing as an error. Here is the latest upload ID 5f8c2f84b5daec0b585d2621

**Comment by calremmel** [comment]
Looking at the logs, the error is being thrown from parsing_lib.py rather from the parser itself:

```
[ERROR] KeyError: 'start'
Traceback (most recent call last):
  File "/var/task/amapa.py", line 180, in lambda_handler
    return parsing_lib.run_lambda(event, context, parse_cases)
  File "/opt/python/parsing_lib.py", line 307, in run_lambda
    common_lib.complete_with_error(
  File "/opt/python/common_lib.py", line 98, in complete_with_error
    raise exception
  File "/opt/python/parsing_lib.py", line 292, in run_lambda
    count_created, count_updated = write_to_server(
  File "/opt/python/parsing_lib.py", line 120, in write_to_server
    batch = batch_of(cases, cases_batch_size)
  File "/opt/python/parsing_lib.py", line 104, in batch_of
    batch.append(next(cases))
  File "/opt/python/parsing_lib.py", line 230, in <genexpr>
    return (case for case in case_data if case_is_within_range(case, cutoff_date, op))
  File "/opt/python/parsing_lib.py", line 216, in case_is_within_range
    case_date = get_case_date(confirmed_event["dateRange"]["start"])
```

**Comment by z023** [comment]
There's still a problem with Amapa. I triggered a retrieval but there are still no uploads or errors in the parser logs.

**Comment by iamleeg** [comment]
@z023 According to the logs, it successfully imported 0 events. Apparently it found no new cases within the last three days. Looking at the raw data the latest case is from 2020-07-20: have they changed the location of their later data, or just stopped updating the report, maybe?

**Comment by AnyaLindstromBattle** [comment]
Yes I was just looking at the data and the issue is, as Graham says, with the data not the parser. 

I did, however, notice another but separate issue. If I download the data from https://drive.google.com/uc?export=download&id=16JBgPV2Px93iD_NbuDo13b3ye4KGxlcW which is the link we are using the last data is from 2020-07-20 as you said, but if I go to the Amapa source website http://painel.corona.ap.gov.br/ and to Baixar CSV to download the document from the Google Drive, the data looks different. The last date is now 29/09/2020 (so we would still get 0 new events as above as this is still more than 3 days ago), and the column titles are different e.g. Data de Notificacao as opposed to dataNotificacao.

I assume we want to get the data that we are missing between July and September into the database. So for this we will need to: 
1. Get a new link to the more up-to-date data - I will sort this and send it to @z023 when I have it. @iamleeg I don't suppose you have any idea why the google drive link that we provided for ingestion didn't 'keep track' of the added data? I just want to make sure that the link I provide isn't a 'snapshot' of the data at that particular time, if that makes sense. 
2. Change the parser to deal with the new format - this should be quick as it's mostly a column name issue
3. Once that's done and merged, we will have to wait for backfilling to work to get the Amapa data ingested. Once that's done, and assuming the new link keeps up with any data added, this should hopefully work as an automated source. 

Sound good?

**Comment by iamleeg** [comment]
If they changed the Google Drive document we would always have play catch-up with that, given the current architecture of the system (otherwise we'd have to make a virtual browser to visit the source website, follow the download link, then get that URL). What you suggest is a good plan.

**Comment by AnyaLindstromBattle** [comment]
Ok cool. So what happens if they update the document again? Do we need to build in some kind of check every few months to manually go and see if the data has been updated and, if it has, update the URL?

**Comment by iamleeg** [comment]
Theoretically we already have that, from the liveness chart on the curator dashboard.

**Pull request merged**